### PR TITLE
Make delta-standalone jar include ParquetSchemaConverter

### PR DESCRIPTION
**Pull request by zsxwing**
This PR fixes the issue that the delta-standalone jar is missing `ParquetSchemaConverter` by:

- Create a dummy project `standaloneWithoutParquetUtils` to allow `standaloneParquet` depending on the shaded delta-standalone jar.
- Shade `standaloneParquet`.
- `standaloneCosmetic` depends on the shaded `standaloneParquet` which include `ParquetSchemaConverter`.
-  Add `testParquetUtilsWithStandaloneCosmetic` and move tests for `ParquetSchemaConverter` into it to verify we shade the code correctly.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
             ~/.ivy2
             ~/.cache/coursier
             ~/.m2
-          key: build-cache-2-with-scala_${{ matrix.scala }}
+          key: build-cache-3-with-scala_${{ matrix.scala }}
```
Comment (zsxwing) [inline .github/workflows/test.yaml:25]: change the key to re-cache since we made a few dependency changes recently.

File: build.sbt [hunk 2]
```diff
@@ -372,15 +372,15 @@ lazy val hive2Tez = (project in file("hive2-tez")) settings (
...
   .settings(
     name := "delta-standalone",
     commonSettings,
     releaseSettings,
     exportJars := true,
-    Compile / packageBin := (standalone / assembly).value,
+    Compile / packageBin := (standaloneParquet / assembly).value,
     libraryDependencies ++= scalaCollectionPar(scalaVersion.value) ++ Seq(
       "org.apache.hadoop" % "hadoop-client" % hadoopVersion % "provided",
+      "org.apache.parquet" % "parquet-hadoop" % "1.12.0" % "provided",
```
Comment (zsxwing) [inline build.sbt:383]: Adding this back as 0.4.1 has this in pom.xml as a provided dependency.

**Review by scottsand-db — APPROVED** [review]
LGTM

**Comment by scottsand-db** [comment]
Verified Hive jars. Awesome.

**Pull request merged**